File: adventofcode/java/year2016/day05/Day05Test.java
package adventofcode.java.year2016.day05;

import java.io.FileReader;
import java.io.IOException;
import java.io.StringReader;

import org.junit.Test;
import org.junit.Assert;

public class Day05Test {
    @Test
    public void part1Example() {
        var input = new StringReader("abc");
        var output = "18f47a30";
        var result = Day05.part1(input);
        Assert.assertEquals("no error", "", result.error);
        Assert.assertEquals("correct output", output, result.answer);
    }

    @Test
    public void part1Actual() throws IOException {
        try (var input = new FileReader("adventofcode/data/year2016/day05/actual.in")) {
            var output = "c6697b55";
            var result = Day05.part1(input);
            Assert.assertEquals("no error", "", result.error);
            Assert.assertEquals("correct output", output, result.answer);
        }
    }

    @Test
    public void part2Example() {
        var input = new StringReader("abc");
        var output = "05ace8e3";
        var result = Day05.part2(input);
        Assert.assertEquals("no error", "", result.error);
        Assert.assertEquals("correct output", output, result.answer);
    }

    @Test
    public void part2Actual() throws IOException {
        try (var input = new FileReader("adventofcode/data/year2016/day05/actual.in")) {
            var output = "8c35d1ab";
            var result = Day05.part2(input);
            Assert.assertEquals("no error", "", result.error);
            Assert.assertEquals("correct output", output, result.answer);
        }
    }
}

File: adventofcode/cpp/year2019/day05/day05_test.cc
#include "adventofcode/cpp/year2019/day05/day05.h"

#include <fstream>

#include "gtest/gtest.h"

TEST(Year2019Day05, Part1Actual) {
  std::ifstream input("adventofcode/data/year2019/day05/actual.in");
  std::string output = "12896948";
  adventofcode::answer_t a = day05::part1(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
  input.close();
}

TEST(Year2019Day05, Part2Actual) {
  std::ifstream input("adventofcode/data/year2019/day05/actual.in");
  std::string output = "7704130";
  adventofcode::answer_t a = day05::part2(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
  input.close();
}
